### ISSUE TITLE
help on using of LibCallSimplifier 

### ISSUE THREAD

**Issue by ash1852** [llvm/llvm-project#56796]
https://github.com/llvm/llvm-project/blob/999514bb9ae1e0125d2a778abaec1b8dac6d3e6f/llvm/include/llvm/Transforms/Utils/SimplifyLibCalls.h#L97-L100

hello! I am reading the source of a project built on llvm, and there is a use of LibCallSimplifier::optimizeCall(Instruction*), I simply read the annotation of this function and the class it belong to, but I still don't understand what this class and this function do exactly with the instruction.

In short, what exactly does Libcall mean here, or how distinguish between libcalls and other calls, and how does the so-called "optimize" work? I can't understand why transform "printf" to "puts"(in class annotation)...

thanks for your help!

**Comment by fhahn** [comment]
For questions, please use https://discourse.llvm.org
